### ISSUE TITLE
CARLA Architecture

### ISSUE THREAD

**Issue by JohnEmad92** [carla-simulator/carla#145]
Is the software architecture of CARLA available?

**Comment by nsubiron** [comment]
Yes, at [CARLA Design](http://carla.readthedocs.io/en/latest/carla_design/). The documentation is still very minimal, but we are working to make it better.

**Comment by JohnEmad92** [comment]
Thanks for your response,
But there is no detailed design for the modules and how they interact that could be used if I need to develop a new sensor?

**Comment by nsubiron** [comment]
We don't have yet a good way for a adding new sensors sorry, this will be done hopefully when finishing the Lidar support #105.
